File: payzo-mobile/lib/services/wallet_service.dart
import '../models/wallet.dart';
import 'api_service.dart';

class WalletService {
  final ApiService _api;
  WalletService(this._api);

  Future<WalletModel> getWallet() async {
    final res = await _api.get('/wallet');
    return WalletModel.fromJson(res.data['data']);
  }
}

File: payzo-web/app/Services/WalletService.php
<?php

namespace App\Services;

use App\Models\User;
use App\Models\Wallet;

class WalletService
{
    /**
     * Return the authenticated user's wallet.
     */
    public function getWallet(User $user): Wallet
    {
        return $user->wallet;
    }
}
